packages/generator/src/react/context.ts

```diff
@@ -15,9 +15,28 @@ export class DefaultImportBuilder {
         //formattedDateTimeRange.importDeclaration, formattedRelativeTime.importDeclaration, 
         //formattedPlural.importDeclaration, formattedMessage.importDeclaration
     }
-
-    unique() {
-        return this.imports // TODO unique
+    
+    unique() {     
+        const tempImoprtedArr:ts.__String[] = []
+        return this.imports.map(i => {
+            let newNamedImportlist:ts.ImportSpecifier[] = []
+            let isRepeatExists:boolean = false
+            i.importClause?.namedBindings?.forEachChild(child => {
+                if(ts.isImportSpecifier(child)){
+                    if(tempImoprtedArr.indexOf(child.name.escapedText) > -1){
+                        isRepeatExists = true;
+                    }else{                      
+                        tempImoprtedArr.push(child.name.escapedText)
+                        newNamedImportlist.push(child) 
+                    }
+                }
+            })
+            if(!isRepeatExists){
+                return i
+            }else if(isRepeatExists && newNamedImportlist.length > 0){
+                return  factory.updateImportDeclaration(i,i.decorators,i.modifiers,factory.createImportClause(i.importClause!.isTypeOnly,i.importClause!.name,factory.createNamedImports(newNamedImportlist)),i.moduleSpecifier)
+            }
+        }).filter(i => {return i != undefined})
```

Comment:
This functionality will work on all types of symbol imports as you want. If we want it to work on duplicate import names or namespace imports it have to be modified.